packages/serverpod/lib/src/database/concepts/query_mode.dart

```diff
@@ -0,0 +1,8 @@
+/// Options for the Query Execution Mode
+enum QueryMode {
+  /// Extended Query Protocol
+  extended,
+
+  /// Simple Query Protocol
+  simple
+}
```

Comment:
Question: Why is this needed? What is the difference? 

If we are only using this for the migration stuff wouldn't it be better to have a dedicated method for that instead?

Does this affect the generated code in anyway? Do we have to expose this functionality there too?

Reply:
The extended query protocol allows you to supply parameters or create variables in the query, but then it is limited to only a single "statement" performed for each query. We use the extended query protocol when we analyze the database for example, there we create arrays and do calculations in the database as part of the query.

The extended query protocol will also be required if we start using substitutions in Postgres. At the moment we don't use this but it could be used in the future.

The simple query does not allow any substitution or variable declarations but does allow multiple statements to be included in a single query, we use that for applying our migrations.

The driver previously allowed us to always use the extended query protocol but still perform multiple statement. I suspect something is added to the query not that marks it as exclusively extended. Before it used to differentiate between execute and query which means that execute could maybe have been done as a simple query while query was executed as an extended query.

We do need to expose it in the unsafe queries, possibly only in the `execute` method, since our users could need to perform multiple "statements" in a single query, and we default to the extended interface. But we shouldn't need to expose it in our generated code since we control how the query is executed there and we only perform a single query.